### DISCUSSION TITLE
Extended Code 14 error when opening from read-only file system

### DISCUSSION THREAD

**Discussion by Discordanian** [rusqlite/rusqlite#1565] [Questions] [answered]
```
rusqlite 0.32.1
cargo 1.82.0-nightly (257b72b8a 2024-07-30)
```

```
    let conn =
        rusqlite::Connection::open_with_flags(dbfile, rusqlite::OpenFlags::SQLITE_OPEN_READ_ONLY)
            .expect(&errmsg);
```

```
thread 'main' panicked
SqliteFailure(Error { code: CannotOpen, extended_code: 14 }, Some("unable to open database file"))
```

Backstory is that I'm opening a file from a read-only NAS location.   If I manually copy the db files locally where I can read-write, everything works fine.  But I was expecting that with the SQLITE_OPEN_READ_ONLY flag that I wouldn't need read/write to the db file at the OS level.  Am I mistaken?

**Comment by gwenn** [comment, marked as answer]
Maybe
https://sqlite.org/forum/forumpost/42bcda311d3654b3
Or simply try to read it directly with Rust standard API just to make sure that the `dbfile` actually exists.
https://doc.rust-lang.org/std/fs/struct.File.html#method.open

**Reply by Discordanian** [reply]
That works.  If I flip to a URI and add ?immutable=1 I'm able to open from the read-only NAS mount.  Perfect.  Thanks much.
